### ISSUE TITLE
Exception gets indexed when saving/publishing

### ISSUE THREAD

**Issue by skttl** [skttl/umbraco-fulltextsearch8#52]
```
System.NullReferenceException: Object reference not set to an instance of an object. at LightInject.Web.PerWebRequestScopeManager.GetOrAddScope() in C:\projects\lightinject-web\build\tmp\Net46\Binary\LightInject.Web\LightInject.Web.cs:line 148 at DynamicMethod(Object[] ) at Umbraco.Web.Mvc.ContainerControllerFactory.GetControllerInstance(RequestContext requestContext, Type controllerType) in D:\a\1\s\src\Umbraco.Web\Mvc\ContainerControllerFactory.cs:line 21 --- End of inner exception stack trace --- at Umbraco.Web.Mvc.ContainerControllerFactory.GetControllerInstance(RequestContext requestContext, Type controllerType) in D:\a\1\s\src\Umbraco.Web\Mvc\ContainerControllerFactory.cs:line 25 at Umbraco.Web.Mvc.UmbracoControllerFactory.CreateController(RequestContext requestContext, String controllerName) in D:\a\1\s\src\Umbraco.Web\Mvc\UmbracoControllerFactory.cs:line 38 at Umbraco.Web.Mvc.RenderControllerFactory.CreateController(RequestContext requestContext, String controllerName) in D:\a\1\s\src\Umbraco.Web\Mvc\RenderControllerFactory.cs:line 36 at System.Web.Mvc.MvcHandler.ProcessRequestInit(HttpContextBase httpContext, IController& controller, IControllerFactory& factory) at System.Web.Mvc.MvcHandler.ProcessRequest(HttpContextBase httpContext) at Umbraco.Web.Templates.TemplateRenderer.RenderUmbracoRequestToString(RequestContext requestContext) in D:\a\1\s\src\Umbraco.Web\Templates\TemplateRenderer.cs:line 169 at Umbraco.Web.Templates.TemplateRenderer.ExecuteTemplateRendering(TextWriter sw, PublishedRequest request) in D:\a\1\s\src\Umbraco.Web\Templates\TemplateRenderer.cs:line 139 at Umbraco.Web.Templates.TemplateRenderer.Render(Int32 pageId, Nullable`1 altTemplateId, StringWriter writer) in D:\a\1\s\src\Umbraco.Web\Templates\TemplateRenderer.cs:line 108 at Umbraco.Web.UmbracoComponentRenderer.RenderTemplate(Int32 contentId, Nullable`1 altTemplateId) in D:\a\1\s\src\Umbraco.Web\UmbracoComponentRenderer.cs:line 52
``` 

Works as expected when indexed through the dashboard

**Comment by stefankip** [comment]
I see this happening on one of our live websites. Any idea why so I might be able to fix it?

**Comment by SimonAntony** [comment]
I'm getting the same exception on live and dev. I was going to try and grab the source at some point when I get time to see if I could trace it, however clients keep booking me out :(

**Comment by skttl** [comment]
IIRC it got fixed in this 2ded354a34a538f5c942dfd1422796a2f03babd8

All that task.run stuff had some unexpected problems when dealing with lots of content being published at once, so I had to remove it for one of my projects.

Downside to that is, that if the view of a node breaks (YSOD) - then you get an error dialog when publishing with little to no information. (#47)

**Comment by SimonAntony** [comment]
Sounds like you have a good handle on it, nice. Happy to help out testing if you need it and if I can grab some time, i'll do some debugging as well (however maybe 2/3 weeks yet before I can do that)

**Comment by skttl** [comment]
fixed in 2ded354
